In [ ]:
import pandas as pd

# **Read Data**

In [ ]:
monthly_expences_GS = '10OGbaywwMIqKgnPGy8VDvpBVtjyqln47iYa2lFhI9Mw'
monthly_expences = pd.read_excel('https://docs.google.com/spreadsheets/d/' + monthly_expences_GS + '/export?format=xlsx', sheet_name = 'Sheet1')

customer_lifespan_data_GS = '1by8tPHwOnq3uKYK2E7sA9VBUYoPM4p1Rnrm_Ss9cyHI'
customer_lifespan_data = pd.read_excel('https://docs.google.com/spreadsheets/d/' + customer_lifespan_data_GS + '/export?format=xlsx', sheet_name = 'Sheet1')

daily_marketing_spendings_GS = '1AZOIThOV4P-0eYDge53ZwumVkfkHoYPWxst3k3Bv87c'
daily_marketing_spendings = pd.read_excel('https://docs.google.com/spreadsheets/d/' + daily_marketing_spendings_GS + '/export?format=xlsx', sheet_name = 'Sheet1')

Payroll_GS = '1c_WihqTZCQvNgxzmd-OwhR9i5diwtfxXVLyMn8R-Lp4'
Payroll = pd.read_excel('https://docs.google.com/spreadsheets/d/' + Payroll_GS + '/export?format=xlsx', sheet_name = 'Sheet1')

receipts_history_GS = '1qayqML1zCKdmtzutkcy9LWvE6xFRm6TGBEVkHHJKIuE'
receipts_history = pd.read_excel('https://docs.google.com/spreadsheets/d/' + receipts_history_GS + '/export?format=xlsx', sheet_name = 'Sheet1')

# **1. CAC: Customer Acquisition Cost**

**1.1. Monthly Spendings Calculation**

Filtering the last month



In [ ]:
Last_month = monthly_expences['month'].max().month
Last_month

3

In [ ]:
cond = monthly_expences['month'].dt.month == Last_month
last_month_expences = monthly_expences[cond]
last_month_expences

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

Take 'Salesforce' Amount

In [ ]:
cond = last_month_expences['item'] == 'Salesforce'
crm_cost = last_month_expences[cond]['amount'].values[0]
print(crm_cost)

1700


In [ ]:
last_month_expences[cond]['amount']

22    1700
Name: amount, dtype: int64

**1.2. Salaries**

Filtering Last Month

In [ ]:
Last_month_payroll = Payroll['month'].max().month
Last_month_payroll

3

In [ ]:
cond = Payroll['month'].dt.month == Last_month_payroll
last_payroll = Payroll[cond]
last_payroll

month   department   employee_name            position  paid
34 2023-03-01        Sales        John Doe       Sales Manager  1500
35 2023-03-01        Sales      Jane Smith     Sales Associate   600
36 2023-03-01        Sales       Jim Brown     Sales Associate   700
37 2023-03-01        Sales    Laura Miller     Sales Associate   800
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500
41 2023-03-01  Engineering       Diana Lee           Developer  1700
42 2023-03-01  Engineering       Eva White    Junior Developer  1000
43 2023-03-01           HR     Frank Green          HR Manager  1300
44 2023-03-01           HR      Gary Black       HR Specialist   700
45 2023-03-01      Support    Hannah Scott        Support Lead   600
46 2023-03-01      Support      Ian Harris  Support Specialist   500
47 2023-03-01      Support     Emily Clark  Support Specialist   450
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200
50 2023-03-01    Analytics       David Lee        Data Analyst  1300

In [ ]:
cond = last_payroll['department'].isin(["Marketing", "Sales"])
Sales_marketing_payroll = last_payroll[cond]['paid']
print(Sales_marketing_payroll)

34    1500
35     600
36     700
37     800
38    1650
39     700
Name: paid, dtype: int64


In [ ]:
Sales_Marketing_paid = sum(Sales_marketing_payroll)

**1.3 Marketing Costs**

In [ ]:
daily_marketing_spendings.sample(10)

date       channel  spending
284 2023-03-13    Google Ads       871
342 2023-03-27  LinkedIn Ads       985
204 2023-02-21    Google Ads       252
228 2023-02-27    Google Ads       141
55  2023-01-14   Twitter Ads       610
186 2023-02-16  LinkedIn Ads       510
196 2023-02-19    Google Ads       239
289 2023-03-14  Facebook Ads       529
319 2023-03-21   Twitter Ads       268
93  2023-01-24  Facebook Ads       874

In [ ]:
last_month_mkt_expences = daily_marketing_spendings['date'].max().month
last_month_mkt_expences

3

In [ ]:
cond = daily_marketing_spendings['date'].dt.month == last_month_mkt_expences
last_month_mkt = daily_marketing_spendings[cond]['spending']
last_month_mkt

236    449
237    229
238    835
239    986
240    912
      ... 
355    960
356    143
357    183
358    533
359    909
Name: spending, Length: 124, dtype: int64

In [ ]:
last_month_mkt_expences = daily_marketing_spendings[cond]['spending'].sum()
print(last_month_mkt_expences)

68830


**1.5 Total Expences**

In [ ]:
Total_Expences = crm_cost + Sales_Marketing_paid + last_month_mkt_expences
print(Total_Expences)

76480


**1.4. New Customer**

Filter New Customer Last Month

In [ ]:
Last_month_New_Customer = receipts_history['date'].max().month
Last_month_New_Customer

3

In [ ]:
cond = receipts_history['date'].dt.month == Last_month_New_Customer
New_Customer = receipts_history[cond]
New_Customer

date  customer_id  receipt_amount  new_customer
618 2023-03-01         1062             103             0
619 2023-03-01         2243             157             0
620 2023-03-01         1166             372             0
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[311 rows x 4 columns]

In [ ]:
cond = New_Customer['new_customer'] == 1
Last_month_New_Customer = New_Customer[cond]

In [ ]:
Last_month_New_Customer.sample(10)

date  customer_id  receipt_amount  new_customer
898 2023-03-27         1696             431             1
895 2023-03-27         2791             222             1
726 2023-03-11         2961             257             1
920 2023-03-30         2151             482             1
725 2023-03-11         1703              46             1
757 2023-03-13         2851              87             1
805 2023-03-20         1667             434             1
662 2023-03-05         1651             479             1
781 2023-03-17         1047             446             1
673 2023-03-05         1143             405             1

In [ ]:
Numberof_new_Customer = len(Last_month_New_Customer)
Numberof_new_Customer

63

**1.4 Final CAC Calculation**

In [ ]:
CAC = Total_Expences / Numberof_new_Customer
print(CAC)

1213.968253968254


# **2.Average Revenue Per User (ARPU)**

**2.1 Total Revenue**

In [ ]:
Last_month_Revenue = receipts_history['date'].max().month
Last_month_Revenue

3

In [ ]:
cond = receipts_history['date'].dt.month == Last_month_Revenue
Last_month_Revenue = receipts_history[cond]
Last_month_Revenue

date  customer_id  receipt_amount  new_customer
618 2023-03-01         1062             103             0
619 2023-03-01         2243             157             0
620 2023-03-01         1166             372             0
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[311 rows x 4 columns]

In [ ]:
Total_Revenue = Last_month_Revenue['receipt_amount'].sum()
Total_Revenue

np.int64(83033)

**2.2 Total Customer**

In [ ]:
TotalCustomer_last_month = len(New_Customer['customer_id'].unique())
TotalCustomer_last_month

292

**2.3 ARPU Calculation**

In [ ]:
ARPU = Total_Revenue / TotalCustomer_last_month
print(ARPU)

284.3595890410959


# **3. Cost of Goods Sold (COGS)**

Last Month Purchase During Period

In [ ]:
last_month_expences

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
22  23 2023-03-01  Software Licenses            Salesforce    1700
23  24 2023-03-01  Software Licenses                  Zoom     540
24  25 2023-03-01      Office Rental           Office Rent   10000
25  26 2023-03-01              Other       Office Supplies     600
26  27 2023-03-01              Other       Travel Expenses    3200

In [ ]:
production_rows = ['AWS Hosting', 'Google Cloud Storage', 'Atlassian Jira', 'Slack', 'Zoom']
production_purchase = last_month_expences[last_month_expences['item'].isin(production_rows)]
production_purchase

#      month           category                  item  amount
18  19 2023-03-01       Server Costs           AWS Hosting    8400
19  20 2023-03-01       Server Costs  Google Cloud Storage    4400
20  21 2023-03-01  Software Licenses        Atlassian Jira    1400
21  22 2023-03-01  Software Licenses                 Slack     900
23  24 2023-03-01  Software Licenses                  Zoom     540

In [ ]:
solfware_n_server_expences = production_purchase['amount'].sum()
print(solfware_n_server_expences)

15640


Last Month payroll (Engineering, Support)

In [ ]:
last_payroll

month   department   employee_name            position  paid
34 2023-03-01        Sales        John Doe       Sales Manager  1500
35 2023-03-01        Sales      Jane Smith     Sales Associate   600
36 2023-03-01        Sales       Jim Brown     Sales Associate   700
37 2023-03-01        Sales    Laura Miller     Sales Associate   800
38 2023-03-01    Marketing   Alice Johnson   Marketing Manager  1650
39 2023-03-01    Marketing       Bob Davis  Content Specialist   700
40 2023-03-01  Engineering  Charlie Wilson    Senior Developer  2500
41 2023-03-01  Engineering       Diana Lee           Developer  1700
42 2023-03-01  Engineering       Eva White    Junior Developer  1000
43 2023-03-01           HR     Frank Green          HR Manager  1300
44 2023-03-01           HR      Gary Black       HR Specialist   700
45 2023-03-01      Support    Hannah Scott        Support Lead   600
46 2023-03-01      Support      Ian Harris  Support Specialist   500
47 2023-03-01      Support     Emily Clark  Support Specialist   450
48 2023-03-01    Analytics   Michael Brown   Analytics Manager  1700
49 2023-03-01    Analytics    Sarah Wilson        Data Analyst  1200
50 2023-03-01    Analytics       David Lee        Data Analyst  1300

In [ ]:
production_salaries_df = last_payroll[last_payroll['department'] == "Engineering"]
production_salaries_df

month   department   employee_name          position  paid
40 2023-03-01  Engineering  Charlie Wilson  Senior Developer  2500
41 2023-03-01  Engineering       Diana Lee         Developer  1700
42 2023-03-01  Engineering       Eva White  Junior Developer  1000

In [ ]:
production_salaries = production_salaries_df['paid'].sum()
print(production_salaries)

5200


In [ ]:
COGS = solfware_n_server_expences + production_salaries
print(COGS)

20840


# **4. GROSS Margin**

In [ ]:
Last_month_Revenue

date  customer_id  receipt_amount  new_customer
618 2023-03-01         1062             103             0
619 2023-03-01         2243             157             0
620 2023-03-01         1166             372             0
621 2023-03-01         2406             426             1
622 2023-03-01         2761              41             1
..         ...          ...             ...           ...
924 2023-03-31         2079             358             0
925 2023-03-31         2089             375             0
926 2023-03-31         1355             249             0
927 2023-03-31         1062             111             0
928 2023-03-31         1590             271             0

[311 rows x 4 columns]

In [ ]:
Revenue = Last_month_Revenue['receipt_amount'].sum()
print(Revenue)

83033


In [ ]:
GrossMargin = (Revenue - COGS) / Revenue * 100
print(GrossMargin)

74.9015451687883


# **5. Customer Life Time Value (LTV)**

In [ ]:
customer_lifespan_data.sample(10)

Unnamed: 0 start_date churn_date
84        1084 2023-03-18 2023-03-23
8         1008 2022-04-25 2022-08-20
53        1053 2021-03-20 2022-12-07
9         1009 2020-04-25 2021-06-01
2         1002 2022-10-30 2023-02-04
58        1058 2021-03-26 2022-07-18
87        1087 2021-12-31 2022-01-23
41        1041 2021-01-05 2023-03-31
55        1055 2022-02-09 2022-06-03
60        1060 2022-01-19 2022-03-11

In [ ]:
#Handle cases when churn_date is None (Customer did not churn whithin the observed period)
lifespan_data = customer_lifespan_data.dropna(subset=['churn_date'])

In [ ]:
# Calculate customer lifespan in days (in the new column "lifespan_days")
lifespan_data['lifespan_days'] = (lifespan_data['churn_date'] - lifespan_data['start_date']).dt.days

# Calculate the average lifespan in days
avg_lifespan_days = lifespan_data['lifespan_days'].mean()

# Convert average lifespan to months
avg_lifespan_months = avg_lifespan_days / 30

print(avg_lifespan_months)

9.841333333333333


In [ ]:
LTV = ARPU * avg_lifespan_months * GrossMargin/100
print(LTV)

2096.1028904109585


# **6. LTV / CAC Ratio**

In [ ]:
LTV_CAC_Ratio = LTV / CAC
print(LTV_CAC_Ratio)

1.726653793094801


# **7. Conclusion**

So sánh chỉ số LTV (2096 đôla) và CAC (1214 đôla), cho thấy kinh phí cho khách hàng mới đang tốt và đáng để đầu tư.

Mức chỉ số LTV/CAC Ratio đang ở mức trên >1, điều này cho thấy công ty đang có lợi nhuận trong hoạt động kinh doanh.

Công ty nên duy trì các hoạt động ở công ty hiện tại và theo dõi chỉ số này thường xuyên để có thể duy trì lợi nhuận lâu dài cho doanh nghiệp.